tests/core/data_layer/test_data_rpc.py

```diff
@@ -2328,3 +2349,292 @@ async def test_mirrors(
+    with pytest.raises(ValueError, match="Invalid Proof: No DL singleton found at coin id"):
+        await offer_setup.taker.api.verify_proof(request=badproof)
+
+    # test bad innerpuz
+    badproof = deepcopy(proof)
+    badproof["inner_puzzle_hash"] = bytes32.random(seeded_random).hex()
+    with pytest.raises(ValueError, match="Invalid Proof: incorrect puzzle hash"):
+        await offer_setup.taker.api.verify_proof(request=badproof)
+
+    # test bad key
+    badproof = deepcopy(proof)
+    badproof["proof"]["proofs"][0]["key_clvm_hash"] = bytes32.random(seeded_random).hex()
+    with pytest.raises(ValueError, match="Invalid Proof: node hash does not match key and value"):
+        await offer_setup.taker.api.verify_proof(request=badproof)
+
+    # test bad value
+    badproof = deepcopy(proof)
+    badproof["proof"]["proofs"][0]["value_clvm_hash"] = bytes32.random(seeded_random).hex()
+    with pytest.raises(ValueError, match="Invalid Proof: node hash does not match key and value"):
+        await offer_setup.taker.api.verify_proof(request=badproof)
+
+    # test bad layer hash
+    badproof = deepcopy(proof)
+    badproof["proof"]["proofs"][0]["layers"][1]["other_hash"] = bytes32.random(seeded_random).hex()
+    with pytest.raises(ValueError, match="Invalid Proof: invalid proof of inclusion found"):
+        await offer_setup.taker.api.verify_proof(request=badproof)
+
+
+@pytest.mark.limit_consensus_modes(reason="does not depend on consensus rules")
+@pytest.mark.anyio
+async def test_dl_proof_changed_root(offer_setup: OfferSetup, seeded_random: random.Random) -> None:
+    two_key_proof = populate_reference(count=5, keys_to_prove=2)
+    offer_setup = await populate_proof_setup(offer_setup=offer_setup, count=two_key_proof.entries_to_insert)
+    two_key_proof.verify_proof_response["verified_clvm_hashes"]["store_id"] = f"0x{offer_setup.maker.id.hex()}"
+
+    proof = await offer_setup.maker.api.get_proof(
+        request={"store_id": offer_setup.maker.id.hex(), "keys": two_key_proof.keys_to_prove}
+    )
+    assert proof["success"] is True
+
+    verify = await offer_setup.taker.api.verify_proof(request=proof)
+    assert verify == two_key_proof.verify_proof_response
+
+    key = b"a"
+    value = b"\x00\x01"
+    changelist: List[Dict[str, str]] = [{"action": "insert", "key": key.hex(), "value": value.hex()}]
+    await offer_setup.maker.api.batch_update({"id": offer_setup.maker.id.hex(), "changelist": changelist})
+
+    await process_for_data_layer_keys(
+        expected_key=key,
+        expected_value=value,
+        full_node_api=offer_setup.full_node_api,
+        data_layer=offer_setup.maker.data_layer,
+        store_id=offer_setup.maker.id,
+    )
+
+    root_changed = await offer_setup.taker.api.verify_proof(request=proof)
+    assert root_changed["current_root"] is False
+    # comparing everything except "current_root"
+    assert root_changed == {**verify, "current_root": root_changed["current_root"]}
```

Comment:
I guess this would be pretty similar.  It varies due to checking equality rather than identify of the `"current_root"` value.  Is it meaningful to have these checks separate?  Single asserts are general nice.

```suggestion
    assert root_changed == {**verify, "current_root": False}
```